pkg/skaffold/build/buildpacks/buildpacks.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildpacks
+
+import (
+	"context"
+	"io"
+	"os/exec"
+
+	"github.com/pkg/errors"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/misc"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+// BuildpackBuilder is a builder for buildpack artifacts
+type BuildpackBuilder struct {
+	localDocker docker.LocalDaemon
+	pushImages  bool
+}
+
+// NewArtifactBuilder returns a new buildpack artifact builder
+func NewArtifactBuilder(localDocker docker.LocalDaemon, pushImages bool) *BuildpackBuilder {
+	return &BuildpackBuilder{
+		localDocker: localDocker,
+		pushImages:  pushImages,
+	}
+}
+
+// Build builds an artifact with the pack CLI.
+func (b *BuildpackBuilder) Build(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) (string, error) {
+	if !a.BuildpackArtifact.UsePackCLI {
+		panic("Not yet implemented")
+	}
+
+	// This helps for better caching: `pack` uses a digest of the image name
+	// as the cache ID.
+	parsed, err := docker.ParseReference(tag)
+	if err != nil {
+		return "", errors.Wrapf(err, "parsing tag %s", tag)
+	}
+	latest := parsed.BaseName + ":latest"
+
+	cmd, err := b.retrieveCmd(ctx, out, a, latest)
+	if err != nil {
+		return "", errors.Wrap(err, "retrieving cmd")
+	}
+
+	if err := cmd.Start(); err != nil {
+		return "", errors.Wrap(err, "starting cmd")
+	}
+
+	if err := misc.HandleGracefulTermination(ctx, cmd); err != nil {
+		return "", errors.Wrap(err, "waiting for cmd")
+	}
+
+	if err := b.localDocker.Tag(ctx, latest, tag); err != nil {
+		return "", errors.Wrap(err, "tagging")
+	}
+
+	if b.pushImages {
+		return b.localDocker.Push(ctx, out, tag)
+	}
+	return b.localDocker.ImageID(ctx, tag)
+}
+
+func (b *BuildpackBuilder) retrieveCmd(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) (*exec.Cmd, error) {
+	artifact := a.BuildpackArtifact
+
+	args := []string{"build"}
+
+	// Builder
+	builder := artifact.Builder
+	if builder == "" {
+		builder = "heroku/buildpacks"
```

Comment:
My 2¢: in the absence of a blessed CNB builder, we should error and suggest `pack suggest-builders`.

Reply:
You're right